### ISSUE TITLE
USER ISSUE: Server List Not Loading

### ISSUE THREAD

**Issue by Troy8878** [StrangeLoopGames/EcoIssues#5211]
**Version:** 0.7.1.2 beta

**NOTE:** I was/(am?) dealing with the oft-reported "Connection Failed" problem. Tests show that I've successfully forwarded ports 2999-3001 on my router + modem but now I have this issue and don't know how to proceed.

**Steps to Reproduce:**
1. Open game
2. Click Join

**Expected behavior:**
Server list loads.

**Actual behavior:**
Doesn't load any servers but favorites. I can't connect to them so I don't know if they're actually loaded or not.  Manually adding a server address results in an entry that only contains said address as its name and no other info (is listed as offline even when it's not). Single player works fine.

Update I:  Favorited server statistics (players connected, trades made, et cetera) are updating properly, so there's some sort of connection being made, but only to favorited servers with no other connections seeming to occur.

Update II:  The server I had favorited and tried connecting to apparently has had no unsuccessful queries made (accord to: https://ecoservers.io/server/452/stats/) so while I received information from it I don't seem to be able to send any to it.

**Comment by MarjieAVolk** [comment]
Sounds like https://github.com/StrangeLoopGames/EcoIssues/issues/4689 or https://github.com/StrangeLoopGames/EcoIssues/issues/4995

If you send me your log file at 'C:\Users\\[username]\AppData\LocalLow\Strange Loop Games\ECO\output_log.txt' I can confirm if it is the same issue.

Either way, deleting the folder 'C:\Users\\[username]\AppData\LocalLow\Strange Loop Games\ECO' will probably fix the problem for you.

**Comment by MarjieAVolk** [comment]
O.o

**Comment by MarjieAVolk** [comment]
Just to confirm, did you send that output file after experiencing the problem?  I don't see any relevant errors there, but it is cleared each time you launch the game.

**Comment by Troy8878** [comment]
Yes, that output is immediately after experiencing the problem and nothing more.  I just double-checked with a re-install and the output was the same, save for load times and thread ids of course.

**Comment by Troy8878** [comment]
An odd thing happened just now.  I decided to re-validate the game files via Steam with this install and it detected one unvalidated file, but the only change in behavior was that a line (bolded here) got added to the output.txt:
_____________________________________________________
(Filename: C:\buildslave\unity\build\artifacts/generated/common/runtime/DebugBindings.gen.cpp Line: 51)

**Can** **not** **play** **a** **disabled** **audio** **source**
 
(Filename:  Line: 707)
_____________________________________________________
So basically the re-validation only appears to have added an additional error.

**Comment by MarjieAVolk** [comment]
Okay, another idea - Can you start your own server?  If you can't from in game, also try using the Eco Server tool.

**Comment by Troy8878** [comment]
I have been able to use the Eco Server tool and connect to a server run by it, and it showed up as a LAN server.  I don't have any way to test this outside of a LAN setup unfortunately.

**Comment by Troy8878** [comment]
Okay, I was able to connect to a server recently.  Not all of them showed up on the list, but some of them did.  I haven't been able to confirm yet but I'm suspecting that my networking hardware is to blame.

**Comment by Troy8878** [comment]
Tried connecting to more servers, getting the same output.txt as #5292 most of the time, with the next server list query coming up empty and a crash upon exiting being consistent as well.

**Comment by MarjieAVolk** [comment]
Alriiiiight, so you can run a server locally and connect to it.

Okay so you are seeing lots of these now?  "Ping: Error performing ICMP transmission. Possibly because of a timeout"

Has the server list ever worked for you?  It didn't stop working suddenly or anything?  If it has never worked, and you are getting those timeouts in your log, it does sound like a network problem.

Well kind of a goofy idea, but you could try looking up the server info directly in your browser, see if that works.  If it doesn't work, then you definitely have a network problem.  If it does work, it's probably a bug in the game code.

So here's how you would do it.  In your browser, go to https://ecoauth.strangeloopgames.com/game/serverlist.  You should see a very long list like this:
![image](https://user-images.githubusercontent.com/23460210/37007768-d2bcd232-2093-11e8-8941-88fe78230e43.png)
On my machine it loads almost instantly.

Next, pick any one of the servers listed there, and look up its details.  For example if I pick {"ip":"1234","webport":5678}, then I would go to "http://1234:5678/info".  You should see something like this, and again it should load almost instantly:
![image](https://user-images.githubusercontent.com/23460210/37007847-3957e68a-2094-11e8-9332-310ed84f4531.png)

**Comment by Troy8878** [comment]
Sorry for the late reply.  I did end up replacing my network hardware.  I was getting what appeared to be ICMP errors with other services and sites, but the new hardware has resolved all of the issues I was having before (and according to speed tests is running 4x faster than my previous setup, which already could maintain a stable connection to a server when it did decide to work) _except_ for this game.  I still get "Ping: Error performing ICMP transmission. Possibly because of a timeout" multiple times in the output.txt.

To clarify, the server list works on occasion, and I can connect to servers on occasion, but these don't always work at the same time (i.e. server list loading but no ability to connect, or server list not loading but with ability to add a server to favorites via ip + port input to load it and then connect).  Sometimes neither works, and upon further investigation port forwarding doesn't appear to have any actual effect on connectivity for me.

I followed your instructions, both for a random server and for the server I've been trying to contribute to when I am able to connect (206.221.190.10:3001, if it helps at all).  All pages loaded within a second, and that's with the 30-something tabs I normally have open in Opera + Firefox (I have tested the game with no tabs open ~ again no effect on connectivity).

From what you said in your latest post, it does seem to be something to do with the game code rather than my network at this point.

**Comment by MarjieAVolk** [comment]
Okay, thanks for the reply.

Have you experimented with clicking 'Stop' in the server list window?  I'm wondering if maybe your setup can handle talking to one server at a time, but it can't handle talking to all the servers in the list at once (thus the potential timeouts).

During the times when the server list loads but you can't connect - what does that look like exactly?  Is the connect button hidden by an error message?  Or does the connection fail when you click the connect button?

**Comment by rainfarmer** [comment]
user says:
I have a problem ith the game ... yesterday i play few minutes on my favorites server, but after two crash computer, when i launch the game, i cannot see the list of server, 0 server available... and if i add server with the IP. the game say me he is offline but other player are on it !
I try many things, restart the computer, verify integrity file games on steam, stop fire wall and Nod32...
I have always the same problem, i cannot play..

**Comment by rainfarmer** [comment]
+1

**Comment by rainfarmer** [comment]
+1

**Comment by MarjieAVolk** [comment]
I'm guessing that in 7.4, with the new networking code and the new server browser, this problem will no longer happen.  If anyone still has trouble after updating, let me know.
